test/test_torch.py

```diff
@@ -17768,6 +17767,45 @@ def is_view_of(self, base, other):
 
         return True
 
+    @onlyOnCPUAndCUDA
+    def test_views_with_dtype_change(self, device):
+        def do_transpose(x, contiguous=False, dim0=0, dim1=1):
```

Comment:
This function looks a bit weird. In particular because it is called `do_transpose` but does not always do transpose... 
Maybe find a better name?